src/full/Agda/TypeChecking/MetaVars/Occurs.hs

```diff
@@ -263,7 +265,7 @@ metaCheck m = do
     when (mvFrozen mv == Frozen)             $ fail "meta is frozen"
     unless (isOpenMeta $ mvInstantiation mv) $ fail "meta is already solved"
     unless allowAssign                       $ fail "assigning metas is not allowed here"
-    when (isFlexible cxt)                    $ fail "occurrence is flexible"
+    when (isFlexible cxt && not firstOrder)  $ fail "occurrence is flexible"
```

Comment:
Please add a comment here motivating this change.  (Or rather, the resulting logic.)